### ISSUE TITLE
Invalid username, password or secret key. 504 Server Error, even with the test user.

### ISSUE THREAD

**Issue by migrau** [EGA-archive/ega-download-client#169]
pyEGA3 4.0.5

I got an error when downloading any dataset. First I tested using my credentials, but then I realized that the error is the same even when using the test user described in the documentation:

```
$ pyega3 -d -t datasets
[2022-08-25 05:19:08 +0200] 
[2022-08-25 05:19:08 +0200] pyEGA3 - EGA python client version 4.0.5 (https://github.com/EGA-archive/ega-download-client)
[2022-08-25 05:19:08 +0200] Parts of this software are derived from pyEGA (https://github.com/blachlylab/pyega) by James Blachly
[2022-08-25 05:19:08 +0200] Python version : 3.8.10
[2022-08-25 05:19:08 +0200] OS version : Linux #49~20.04.1-Ubuntu SMP Thu Aug 4 19:15:44 UTC 2022
[2022-08-25 05:19:08 +0200] Server URL: https://ega.ebi.ac.uk:8052/elixir/data
[2022-08-25 05:19:08 +0200] Session-Id: 378223183
[2022-08-25 05:19:08 +0200] Starting new HTTPS connection (1): ipinfo.io:443
[2022-08-25 05:19:08 +0200] https://ipinfo.io:443 "GET /json HTTP/1.1" 200 None
[2022-08-25 05:19:08 +0200] Starting new HTTPS connection (1): ega.ebi.ac.uk:8443
[2022-08-25 05:20:10 +0200] https://ega.ebi.ac.uk:8443 "POST /ega-openid-connect-server/token HTTP/1.1" 504 168
[2022-08-25 05:20:10 +0200] 
[2022-08-25 05:20:10 +0200] Invalid username, password or secret key - please check and retry. If problem persists contact helpdesk on helpdesk@ega-archive.org
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/pyega3/libs/auth_client.py", line 37, in token
    r.raise_for_status()
  File "/usr/local/lib/python3.8/dist-packages/requests/models.py", line 953, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 504 Server Error: Gateway Time-out for url: https://ega.ebi.ac.uk:8443/ega-openid-connect-server/token
```

I tested the ports with
```
openssl s_client -connect ega.ebi.ac.uk:8443
openssl s_client -connect ega.ebi.ac.uk:8052
```
And both are opened. 

I tried installing pyEGA3 with conda and pip3 with the same result.

**Comment by nicoleversetwo** [comment]
Same issues. Been ongoing for a while now.

**Comment by CRMacPherson** [comment]
Same issue.

**Comment by aurelieGabriel** [comment]
I am having the same issue.

**Comment by jason2389** [comment]
I am also having the same issue

**Comment by jason2389** [comment]
Did anybody get an answer from EGA? We would need quite urgently the datasets and I cannot find any solution to this problem. 

Thanks!

**Comment by migrau** [comment]
They told me that the dev team is aware and working on this, as well as the release of a new data API that is more stable. Update expected in a week(s)

**Comment by mimifp** [comment]
I have the same issue. I wrote to them but I have not get answer yet.

**Comment by malloryfreeberg** [comment]
Hi all. I’m sorry you are experiencing this issue, and thank you for reporting it. We are aware of the instability of the service at the moment and are actively working to solve the underlying intermittent issues. If you continue to experience download issues, the best place to get assistance is by emailing the EGA Helpdesk ([helpdesk@ega-archive.org](mailto:helpdesk@ega-archive.org)) and attaching the output log produced by pyega3 (pyega3_output.log). The Helpdesk team can then provide further assistance and updates on a fix. Thank you in advance for your patience while we work to resolve this!

**Comment by christineyyeh** [comment]
Having the same issue here; has there been an update?

**Comment by malloryfreeberg** [comment]
Thank you, all, for your patience while we worked on and released a new version of both [pyega3 (v5.0.0)](https://pypi.org/project/pyega3/) and the underlying API. Please update to the newest version of pyega3 (you can find the instructions [here](https://github.com/EGA-archive/ega-download-client#installation-and-update)) to enjoy the new features, including a more stable service which results in a higher download success rate. You should also only receive the "Invalid username, password or secret key" error if indeed the wrong credentials are being used.

This GH issue will be closed, but if you continue to experience the same problem, feel free to re-open it. If you experience any other problems, please email the EGA Helpdesk at [helpdesk@ega-archive.org](mailto:helpdesk@ega-archive.org) and attach the output log produced by pyega3 (pyega3_output.log). The Helpdesk team can then provide further assistance. Thank you!
